scylla/src/transport/connection.rs

```diff
@@ -772,11 +772,50 @@ impl Connection {
 
     pub(crate) async fn batch_with_consistency(
         &self,
-        batch: &Batch,
+        init_batch: &Batch,
         values: impl BatchValues,
         consistency: Consistency,
         serial_consistency: Option<SerialConsistency>,
     ) -> Result<QueryResult, QueryError> {
+        let batch = {
+            let mut batch: Batch = Default::default();
+            batch.config = init_batch.config.clone();
+
+            let mut to_prepare = HashSet::<usize>::new();
+
+            {
+                let mut value_iter = values.batch_values_iter();
+                for (id, stmt) in init_batch.statements.iter().enumerate() {
+                    let value = value_iter.next_serialized().transpose()?;
```

Comment:
Serializing values for a query can be relatively expensive, so it would be best to avoid it. Let's use `skip_next` instead of `next_serialized` in case of `BatchStatement::PreparedStatement`.

Reply:
Done